app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -375,21 +375,28 @@ public void onAppUpdateEvent(AppUpdateEvent event) {
      * Sets up the x,y coordinates of the arc using the trainer level, stores it in Data.arcX/arcY
      */
     private void setupArcPoints() {
-        final int indices = Math.min((int) ((trainerLevel + 1.5) * 2) - 1, 79);
+        /*
+         * Pokemon levels go from 1 to trainerLevel + 1.5, in increments of 0.5.
+         * Here we use dblLevelShifted for levels that are doubled and shifted by - 2; after this adjustment,
+         * the level can be used to index CpM, arcX and arcY.
+         */
+        int maxPokeDblLevelShifted = trainerLevel * 2 + 1; //= (trainerLevel + 1.5) * 2 - 2
+        final int indices = Math.min(maxPokeDblLevelShifted + 1, 79);
```

Comment:
Rename this variable to size. Then just use convertLevelToIndex(trainerlevel + 1.5) and add 1 to it.

Adding 1 because arrays go from 0 to size - 1. <=== This part too me a while to realize last week when reading the code. It was confusing as heck.